swap/src/bitcoin/wallet.rs

```diff
@@ -211,7 +213,24 @@ impl Wallet {
         self.client.lock().await.status_of_script(tx)
     }
 
-    pub fn subscribe_to(&self, tx: impl Watchable + Send + 'static) -> Subscription {
+    pub async fn subscribe_to(&self, tx: impl Watchable + Send + 'static) -> Subscription {
+        if let Some(sub) = self
+            .client
+            .lock()
+            .await
+            .subscriptions
+            .get(&WatchableTransaction {
+                txid: tx.id(),
+                script: tx.script(),
+            })
+        {
+            return Subscription {
+                receiver: sub.clone(),
+                finality_confirmations: self.finality_confirmations,
+                txid: tx.id(),
+            };
```

Comment:
I think you could also just `clone` the full `Subscription` struct instead of storing just the `Receiver`.